lti_consumer/lti_1p1/contrib/django.py

```diff
@@ -0,0 +1,125 @@
+    to XBlock code and functionality. It is recommended that this remain in use
+    until LTI1.3 support is merged, or a better means of loading the template is
+    made available.
+
+    Arguments:
+        html_element_id (string):  Value to use as the HTML element id in the HTML form
+        lti_launch_url (string):  The URL to send the LTI Launch request to
+        oauth_key (string):  The OAuth consumer key
+        oauth_secret (string):  The OAuth consumer secret
+        resource_link_id (string):  Opaque identifier guaranteed to be unique
+            for every placement of the link
+        user_id (string):  Unique value identifying the user
+        roles (string):  A comma separated list of role values
+        context_id (string):  Opaque identifier used to uniquely identify the
+            context that contains the link being launched
+        context_title (string):  Plain text title of the context
+        context_label (string):  Plain text label for the context
+        result_sourcedid (string):  Indicates the LIS Result Identifier (if any)
+            and uniquely identifies a row and column within the Tool Consumer gradebook
+        person_sourcedid (string):  LIS identifier for the user account performing the launch
+        person_contact_email_primary (string):  Primary contact email address of the user
+        outcome_service_url (string):  URL pointing to the outcome service. This
+            is required if the Tool Consumer is accepting outcomes for launches
+            associated with the resource_link_id
+        launch_presentation_locale (string):  Language, country and variant as
+            represented using the IETF Best Practices for Tags for Identifying
+            Languages (BCP-47)
+        custom_parameters (dict): Contains any other keyword arguments not listed
+            above. It will filter out all arguments provided that do not start with
+            'custom_' and will submit the remaining arguments on the LTI Launch form
+
+    Returns:
+        unicode: HTML template with the form and JavaScript to automatically
+            launch the LTI embedding
+    """
+    lti_consumer = LtiConsumer1p1(lti_launch_url, oauth_key, oauth_secret)
+
+    # Set LTI parameters from kwargs
+    lti_consumer.set_user_data(
+        user_id,
+        roles,
+        result_sourcedid,
+        person_sourcedid=person_sourcedid,
+        person_contact_email_primary=person_contact_email_primary
+    )
+    lti_consumer.set_context_data(
+        context_id,
+        context_title,
+        context_label
+    )
+
+    if outcome_service_url:
+        lti_consumer.set_outcome_service_url(outcome_service_url)
+
+    if launch_presentation_locale:
+        lti_consumer.set_launch_presentation_locale(launch_presentation_locale)
+
+    lti_consumer.set_custom_parameters(
+        {
+            key: value
+            for key, value in custom_parameters.items()
+            if key.startswith('custom_')
```

Comment:
Why do the custom parameters have to start with `custom_`? Is that a requirement of the spec?

Reply:
I was following the specifications outlined [here](https://github.com/edx/xblock-lti-consumer/blob/98d4262c5d6dab7a183e029a48822bc9104e6022/lti_consumer/lti_consumer.py#L811-L841) in the original code.

The implementation guide for LTI1.1 is linked below, and if you do a page search for `custom_keyname`. I am having a bit of difficulty understanding whether or not the prefix is required, but it certainly seems to be recommended at the least.

https://www.imsglobal.org/specs/ltiv1p1/implementation-guide

Reply:
Thanks. That seems clear enough.